lib/Canto/Controller/Curs.pm

```diff
@@ -342,6 +345,8 @@ sub _metagenotype_flags
   my $rs = $schema->resultset('Organism');
 
   my $has_host = 0;
+  my $has_pathogen_genes = 0;
```

Comment:
Strictly speaking, checking for pathogen genes shouldn't be necessary, since PHI-Canto forbids starting a session if you don't have any pathogen genes. It allows a session with no hosts, but not the contrary.

I first thought that the only reason to handle the lack of pathogens genes is if it were possible (or will be possible later) to remove the pathogen genes when you're not on the confirmation page &ndash; however, it turns out the fix for escaping the organism confirmation page (#2102) doesn't cover the case where the user hasn't added pathogen genes, so it's still possible to end up on the summary page with no pathogen genes, which makes this fix necessary (although we should really have a fix for the case I've described too).

Given that you've already added the logic, it's probably not worth taking it out, since it does future-proof us against workflow bugs, plus any future need to allow sessions without pathogen genes.